doc/elements-tx-format.md

```diff
@@ -0,0 +1,562 @@
+# Elements Transaction Format
+
+This document describes the format used to serialize Elements transactions. Once a transaction has been converted into this raw, serialized form, it can be broadcast across the network.
+
+This document assumes some familiarity with Bitcoin and Elements (UTXOs, [Script](https://en.bitcoin.it/wiki/Script), assets, pegins, etc.). For more information on those, please refer to the [Bitcoin Wiki](https://en.bitcoin.it/wiki/Main_Page) and the [Elements Code Tutorial](https://elementsproject.org/elements-code-tutorial/overview).
+
+### Data Types
+
+*Notes*:
+* Fields in the following table are listed in the same order in which they are serialized.
+
+* The *Encoding* column in the following tables indicates how the fields are serialized.
+
+* Values are defined in multiples of 100,000,000 (this corresponds directly to how Bitcoin encodes value, where 1 BTC is divisible into 100,000,000 Satoshis). For example, a `ConfidentialAmount` representing 7 units of something (`L-BTC`, inflation keys, etc.) would be encoded as 700,000,000.
```

Comment:
@apoelstra correct me if I'm wrong, but how the element/green UI handles units is ... more complicated than this suggests.  There is something about the asset registry etc. etc.

I think we should simply note that all unit values are in satoshis, or say all values are in the smallest indivisible unit (i.e. satoshis) or something like that.

Reply:
Agreed.

Even if other assets didn't have "natural units" that differed from 10^8 I don't know what 10^8 would have to do with the encoding, so we should drop any mention of it.

Reply:
The reason I mentioned this was because `Example 2` below contains an asset issuance with the number of inflation keys encoded as:
```
0x29b92700 = 700 000 000 
```

Does this correspond to 7 keys? A single inflation key is indivisible, correct?

Reply:
I think that's 700 million reissuance keys.

Reply:
I see. So when I parse the full issuance transaction with `$elements-cli decoderawtransaction <...>`, the result contains:
```
"issuance": {
        "assetBlindingNonce": "0000000000000000000000000000000000000000000000000000000000000000",
        "assetEntropy": "503f1db03b8ed61ace8fbadc2fcb6c200bdf1ada403d74b891e55e1a047b6596",
        "isreissuance": false,
        "token": "6e88fa86a097b9ede564f67c60e8c5d24ed1f10eeb56c3a60223961f2e38737f",
        "asset": "d42c5ba3724b7f7a045968e0c04e69f81070aca8b6e2db88cd84c65dd07b8324",
        "assetamount": 33.00000000,
        "tokenamount": 7.00000000
}
```
      
Is the `"tokenamount": 7.00000000` indicating that there are really 700,000,000 reissuance tokens?

Reply:
lol. yes.

Reply:
Ok that makes sense now. Thanks!

Reply:
I opened https://github.com/ElementsProject/elements/issues/1152 about this.